applications/GeoMechanicsApplication/python_scripts/geomechanics_T_solver.py

```diff
@@ -0,0 +1,157 @@
+            "temperature_absolute_tolerance": 1.0e-9,
+            "residual_relative_tolerance": 1.0e-4,
+            "residual_absolute_tolerance": 1.0e-9,
+            "desired_iterations"         : 4,
+            "max_radius_factor"          : 20.0,
+            "min_radius_factor"          : 0.5,
+            "max_iterations"             : 15,
+            "min_iterations"             : 6,
+            "number_cycles"              : 5,
+            "increase_factor"            : 2.0,
+            "reduction_factor"           : 0.5,
+            "calculate_reactions"        : true,
+            "max_line_search_iterations" : 5,
+            "first_alpha_value"          : 0.5,
+            "second_alpha_value"         : 1.0,
+            "min_alpha"                  : 0.1,
+            "max_alpha"                  : 2.0,
+            "line_search_tolerance"      : 0.5,
+            "rotation_dofs"              : false,
+            "block_builder"              : true,
+            "search_neighbours_step"     : false,
+            "linear_solver_settings":{
+                "solver_type": "AMGCL",
+                "tolerance": 1.0e-6,
+                "max_iteration": 100,
+                "scaling": false,
+                "verbosity": 0,
+                "preconditioner_type": "ILU0Preconditioner",
+                "smoother_type": "ilu0",
+                "krylov_type": "gmres",
+                "coarsening_type": "aggregation"
+            },
+            "problem_domain_sub_model_part_list": [""],
+            "processes_sub_model_part_list": [""],
+            "body_domain_sub_model_part_list": [""],
+            "loads_sub_model_part_list": [],
+            "loads_variable_list": []
+        }""")
+
+        this_defaults.AddMissingParameters(super().GetDefaultParameters())
+        return this_defaults
+
+    def PrepareModelPart(self):
+        super().PrepareModelPart()
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "Model reading finished.")
+
+    def AddDofs(self):
+        KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.TEMPERATURE, self.main_model_part)
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "DOFs added correctly.")
+
+    def Initialize(self):
+        KratosMultiphysics.Logger.PrintInfo("::[GeoMechanics_T_Solver]:: ", "Started Initialization")
+        super().Initialize()
+
+        # Check if everything is assigned correctly
+        self.Check()
+
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "Solver initialization finished and validated.")
+
+    def _ConstructScheme(self, scheme_type, solution_type):
+
+        self.main_model_part.ProcessInfo.SetValue(KratosGeo.DT_TEMPERATURE_COEFFICIENT, 1.0)
+
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver, solution_type", solution_type)
+        if solution_type.lower() == "transient-heat-transfer" or solution_type.lower() == "transient_heat_transfer":
```

Comment:
Perhaps it's just nitpicking, but I would still like to bring this up. I would prefer to have only one acceptable value for this case (`"transient_heat_transfer"`), not two (`transient_heat_transfer` **and** `transient-heat-transfer`). The reason is that having a single string makes it very clear what to use, instead of letting the user decide which value (s)he likes more. Since this is new functionality (i.e. there are no users yet with different kinds of input files), I would propose to limit the supported values of `solution_type` for thermal analysis to `transient_heat_transfer` only. I am aware that the other solvers also offer this "flexibility", but that does not necessarily mean that we must offer it here as well. In fact, by letting the user decide, we make it more difficult for ourselves to write code that uses this input, since the code must make sure to check for both values, which is error-prone. Now it is still rather painless to remove one of them.

Reply:
The keyword with "-" is removed. Unfortunately, such keywords are also in other geomechanics solvers. A suggestion for the future to a create a function that converts "-" into "_" before performing the checks. For example, after reading info from json files.